### PR TITLE
feat: post-render ballot qr codes

### PR DESCRIPTION
**Pull request by beausmith**
Please review this carefully to determine if there is a better way to do this.

### CODE REVIEW

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 1]
```diff
@@ -33,6 +34,38 @@ import QRCode from './QRCode'
 import Prose from './Prose'
 import Text from './Text'
 
+const qrCodeTargetClassName = "qr-code-target"
+
+class PagedQRCodeInjector extends Handler {
+  afterRendered(pages: any) {
+    pages.forEach((page: any) => {
+      const { pageNumber } = page.element.dataset
```
Comment (benadida) [inline client/src/components/HandMarkedPaperBallot.tsx:42]: can't seem to easily find this in the documentation but... you're sure this is 1-indexed?

Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:42]: Yessir…

image

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 2]
```diff
@@ -315,30 +348,12 @@ const HandMarkedPaperBallot = ({
...
-                  precinctId,
-                  ballotStyleId,
-                  pageNumber: 2,
-                  pageCount: 2,
-                })}
-              />
-            </PageFooterQRCode>
+            <PageFooterQRCode
+              className={qrCodeTargetClassName}
+              data-is-live-mode={isLiveMode}
```
Comment (eventualbuddha) [inline client/src/components/HandMarkedPaperBallot.tsx:353]: If `isLiveMode` is `false`, then the `dataset` will be `{ isLiveMode: "false" }`, and `!!"false"` is `true`. I think you'll need to fix that.

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 3]
```diff
@@ -33,6 +34,38 @@ import QRCode from './QRCode'
 import Prose from './Prose'
 import Text from './Text'
 
+const qrCodeTargetClassName = "qr-code-target"
+
+class PagedQRCodeInjector extends Handler {
+  afterRendered(pages: any) {
```
Comment (eventualbuddha) [inline client/src/components/HandMarkedPaperBallot.tsx:40]: (nice to have) Types?

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 4]
```diff
@@ -33,6 +34,38 @@ import QRCode from './QRCode'
 import Prose from './Prose'
 import Text from './Text'
 
+const qrCodeTargetClassName = "qr-code-target"
```
Comment (eventualbuddha) [inline client/src/components/HandMarkedPaperBallot.tsx:37]: I'm surprised prettier isn't making these single quotes.

Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:37]: 🤦 Just realizing that I have to install all the VSCode extensions in the VM for them to work. Will commit linted changes.

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 3 above)
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:40]: I want types. Whats the best way to do this? Can you give me some pointers here? PagedJS itself is not Typescript.

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 2 above)
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:355]: fixed to `data-is-live-mode={isLiveMode === 'true}`

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 3 above)
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx]: Found the "infer from usage" option in VS Code to create types for the variables used. Thanks for the push.

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 5]
```diff
@@ -33,6 +34,49 @@ import QRCode from './QRCode'
 import Prose from './Prose'
 import Text from './Text'
 
+const qrCodeTargetClassName = 'qr-code-target'
+
+interface PagedJSPage {
+  element: {
+    dataset: {
+      pageNumber: any
```
Comment (eventualbuddha) [inline client/src/components/HandMarkedPaperBallot.tsx:42]: From your screenshot it looked like this would be `string`.

Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:42]: D'oh! Yes.

**Pull request merged**